src/vm/dllimport.cpp

```diff
@@ -5913,37 +5913,49 @@ static HMODULE LocalLoadLibraryHelper( LPCWSTR name, DWORD flags, LoadLibErrorTr
 
     HMODULE hmod = NULL;
 
+    AppDomain* pDomain = GetAppDomain();
+
+    hmod = pDomain->FindUnmanagedImageInCache(name);
```

Comment:
Could you use cache/insert into cache in LoadLibraryModuleHierarchy/LoadLibraryModule?
 For some cases, see https://github.com/dotnet/coreclr/blob/246ae784b826c79098a1b3967083c66187786197/src/vm/dllimport.cpp#L5962 and appx cases, these don't call LocalLoadLibraryHelper

Reply:
I moved reading and writing to cache higher so all of the cases can use caching